redbeams/src/main/java/com/sequenceiq/redbeams/service/stack/RedbeamsCreationService.java

```diff
@@ -65,8 +83,32 @@ public DBStack launchDatabaseServer(DBStack dbStack) {
 
         DBStack savedStack = dbStackService.save(dbStack);
 
+        registerDatabaseServerConfig(savedStack);
+
         flowManager.notify(RedbeamsProvisionEvent.REDBEAMS_PROVISION_EVENT.selector(),
                 new RedbeamsEvent(RedbeamsProvisionEvent.REDBEAMS_PROVISION_EVENT.selector(), dbStack.getId()));
         return savedStack;
     }
+
+    private void registerDatabaseServerConfig(DBStack dbStack) {
+        DatabaseServer databaseServer = dbStack.getDatabaseServer();
+        DatabaseServerConfig dbServerConfig = new DatabaseServerConfig();
+
+        dbServerConfig.setResourceStatus(ResourceStatus.SERVICE_MANAGED);
+        dbServerConfig.setAccountId(databaseServer.getAccountId());
+        dbServerConfig.setName(dbStack.getName());
+        dbServerConfig.setEnvironmentId(dbStack.getEnvironmentId());
+        dbServerConfig.setConnectionDriver(dbStack.getDatabaseServer().getConnectionDriver());
+        // username and password are set during conversion to DBStack
+        dbServerConfig.setConnectionUserName(dbStack.getDatabaseServer().getRootUserName());
+        dbServerConfig.setConnectionPassword(dbStack.getDatabaseServer().getRootPassword());
+        dbServerConfig.setDatabaseVendor(dbStack.getDatabaseServer().getDatabaseVendor());
+        dbServerConfig.setDbStack(dbStack);
+        // host and port are set after allocation is complete
+        dbServerConfig.setHost(null);
```

Comment:
Is this necessary? Does it not default to null?

Reply:
Probably not. I have further changes to make, so I'll drop these at that point.